src/components/home.js

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Link } from "react-router-dom";
+import { DeleteApiAction, GetApiAction } from "../redux/action/action";
+
+const Home = () => {
+  const dispatch = useDispatch();
+  const responseData = useSelector((state) => state.reducer.details); //reducer variable name in rootReducer or details
+
+  const handleDelete = (id) => {
+    dispatch(DeleteApiAction(id));
+    window.location["reload"]();
+  };
+
+  useEffect(() => {
+    dispatch(GetApiAction());
+  }, [dispatch]);
+
+  console.log("HOME ", responseData);
+  const result = responseData?.map((data, index) => (
+    <tr key={index}>
+      <th scope="row"> {data.id}</th>
```

Comment:
why do we have to create separate function to render jsx inside a loop ?
You can directly write in the same return .